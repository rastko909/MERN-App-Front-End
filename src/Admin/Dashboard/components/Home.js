
import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import CssBaseline from '@material-ui/core/CssBaseline';

import NavBar from './NavBar';
import SideBar from './SideBar';

import ClaimsList from './ClaimsList';
import BusinessesList from './BusinessesList';

import ViewClaim from './ViewClaim';
import ViewBusiness from './ViewBusiness';

import NewBusiness from './NewBusiness';

const useStyles = makeStyles(theme => ({
  root: {
    display: 'flex',
  },
  content: {
    flexGrow: 1,
    padding: theme.spacing(3),
  },
  toolbar: theme.mixins.toolbar,
}));

export default function Home({view, functions}) {
  const classes = useStyles();
  console.log("view.name is set to:", view.name);

  return (
    <div className={classes.root}>
      <CssBaseline />
      <NavBar functions={functions} />
      <SideBar view={view} functions={functions} />

      <main className={classes.content}>
        <div className={classes.toolbar} />
        {view.name === "openclaims" && <ClaimsList view={view} functions={functions} />}
        {view.name === "businesses" && <BusinessesList view={view} functions={functions} />}
        {view.name === "viewclaim" && <ViewClaim view={view} functions={functions} />}
        {view.name === "viewbusiness" && <ViewBusiness view={view} functions={functions} />}
        {/* {view.name === "newbusiness" && <NewClaim view={view} functions={functions} />} */}
        {view.name === "newbusiness" && <NewBusiness view={view} functions={functions} />}
      </main>
       
    </div>
  );
}