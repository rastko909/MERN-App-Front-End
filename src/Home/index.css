

.home-container {
  width: 100%;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  height: calc(100vh - 64px);
  /* background: #231F20 !important; */
}


.left {
  /* border: 1px solid red; */
  height: 100% !important;
  display: flex;
  flex-direction: column;
  justify-content: space-evenly;
  align-items: center;
  width: 40%;
  border-radius: 0.5rem;
  /* background: white; */
}

.right {
  /* border: 1px solid red; */
  height: 100% !important;
  display: flex;
  flex-direction: column;
  justify-content: space-evenly;
  align-items: center;
  border-radius: 0.5rem;
  /* background: white; */
  width: 50%;
}

.logo-image {
  width: 10% !important;
}

.image1 {
  width: 300px;
  margin-bottom: 10px;
}

.image {
  width: 300px;
}

.home-dialog {
  padding: 1rem;
  width: 80%;
  height: 80vh;
  border-radius: 0.25rem;
  display: flex;
  flex-direction: row;
  justify-content: space-evenly;
  align-items: center;
  /* background: #F9DE48 !important; */
  /* background: #135E89 !important;   */
}

.buttons {
  width: 100% !important;
  height: 50% !important;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-around;
  width: 80%;
  height: 80%;
}

.button {
  width: 100% !important;
  height: 50px;
  padding: 1rem;
  background: #F9DE48 !important;
  color: black !important;
  width: 40%;
  margin: 1rem;
}