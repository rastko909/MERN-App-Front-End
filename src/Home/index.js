import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/styles';
import Paper from '@material-ui/core/Paper';
import Button from '@material-ui/core/Button';
import NavBar from './components/NavBar';
import HelpIcon from '@material-ui/icons/Help';
import PersonPinIcon from '@material-ui/icons/PersonPin';
import ListAltIcon from '@material-ui/icons/ListAlt';
import './index.css';

const styles = (theme) => ({
  button: {
    // margin: this.props.theme.spacing(1),
  },
  input: {
    display: 'none',
  },
});

class Home extends React.Component {

  newClaim = () => {
    this.props.history.push('/claim/new');
  }

  viewClaim = () => {
    this.props.history.push('/claim');
  }


  render = () => {
    // const { classes } = this.props;
    return (
      <>
        <NavBar />
        <div className="home-container">
        <div className="home-inner-container">
        <div className="disclosure-text">
         <h1>
          Welcome to <strong>Disclosure Platform</strong>
         </h1>

         <div className="description-text">
           <div className="description-header">
           <PersonPinIcon /><h3>Who are we?</h3>
           </div>

           <p>Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.</p>
         </div>

         <div className="description-text">
           <div className="description-header">
           <HelpIcon /><h3>How can we help you?</h3>
           </div>
           <p>Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.</p>
         </div>

         <div className="description-text">
         <div className="description-header">
           <ListAltIcon /><h3>Have a claim?</h3>
           </div>
           <p>Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.</p>
         </div>

      <div className="buttons">

         <Button className='button' onClick={this.newClaim}>
            Lodge New Claim
        </Button>

        <Button className='button' onClick={this.viewClaim}>
          Login to Existing Claim
        </Button>
      </div>
        </div>

        </div>


        </div>
      </>
    );
  }
}

Home.propTypes = {
  classes: PropTypes.object.isRequired,
};

export default withStyles(styles)(Home);