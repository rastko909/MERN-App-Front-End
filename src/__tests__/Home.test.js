import React from 'react';
import Home from '../Home/index';
import { create } from 'react-test-renderer'

describe('Home page snapshot test',()=>{
    test('testing homepage', () => {
    let tree = create(<Home  />)
    expect(tree.toJSON()).toMatchSnapshot();
  })
})