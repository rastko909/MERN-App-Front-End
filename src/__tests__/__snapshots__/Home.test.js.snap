// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page snapshot test testing homepage 1`] = `
Array [
  <div
    className="makeStyles-root-3"
  >
    <header
      className="MuiPaper-root MuiPaper-elevation4 MuiAppBar-root MuiAppBar-positionSticky MuiAppBar-colorPrimary"
    >
      <div
        className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
      >
        <button
          aria-label="Menu"
          className="MuiButtonBase-root MuiIconButton-root makeStyles-menuButton-4 MuiIconButton-colorInherit MuiIconButton-edgeStart"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
              />
            </svg>
          </span>
        </button>
        <h6
          className="MuiTypography-root makeStyles-title-5 MuiTypography-h6"
        >
          <i
            className="fas fa-hands-helping icon"
          />
          Disclosure Platform
        </h6>
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            About Us
          </span>
        </button>
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            Your Rights
          </span>
        </button>
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            FAQ
          </span>
        </button>
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            Contact
          </span>
        </button>
      </div>
    </header>
  </div>,
  <div
    className="home-container"
  >
    <div
      className="MuiPaper-root MuiPaper-elevation1 home-dialog MuiPaper-rounded"
    >
      <div
        className="left"
      >
        <button
          className="MuiButtonBase-root MuiButton-root button MuiButton-text"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            Lodge New Claim
          </span>
        </button>
        <img
          alt="logo"
          className="image1"
          src={
            Object {
              "ReactComponent": Object {
                "$$typeof": Symbol(react.forward_ref),
                "render": [Function],
              },
              "__esModule": true,
              "default": "new-claim.svg",
            }
          }
        />
      </div>
      <div
        className="right"
      >
        <button
          className="MuiButtonBase-root MuiButton-root button MuiButton-text"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            Login to Existing Claim
          </span>
        </button>
        <img
          alt="logo"
          className="image"
          src={
            Object {
              "ReactComponent": Object {
                "$$typeof": Symbol(react.forward_ref),
                "render": [Function],
              },
              "__esModule": true,
              "default": "view-claim.svg",
            }
          }
        />
      </div>
    </div>
  </div>,
]
`;
